mrmustard/physics/ansatze.py

```diff
@@ -380,3 +380,114 @@ def __and__(self, other: PolyExpAnsatz) -> PolyExpAnsatz:
         bs = [math.concat([b1, b2], axis=-1) for b1 in self.b for b2 in other.b]
         cs = [math.outer(c1, c2) for c1 in self.c for c2 in other.c]
         return self.__class__(As, bs, cs)
+
+class ArrayAnsatz(Ansatz):
+    r"""Represents the ansatz as a multidimensional array.
+    """
+    def __init__(self, array: Batch[Tensor]):
+        r"""Note that the array is batched."""
+        self.array = array
+    
+    @property
+    def array(self) -> Batch[Tensor]:
+        r"""
+        The array.
+        """
+        return self.array
+
+    def __neg__(self) -> ArrayAnsatz:
+        r"""
+        Negates the values in the array.
+        """
+        return self.__class__(array=-self.array)
+
+    def __eq__(self, other: Ansatz) -> bool:
+        r"""
+        Whether this ansatz's array is equal to another ansatz's array.
+
+        Note that the comparaison is done by numpy allclose with numpy's default rtol and atol.
+
+        #TODO: make sure this is the correct type of the wrong shape
+        """
+        try:
+            return np.allclose(self.array, other.array)
+        except Exception as e:
+            raise TypeError(f"Cannot add {self.__class__} and {other.__class__}.") from e
+
+    def __add__(self, other: ArrayAnsatz) -> ArrayAnsatz:
+        r"""
+        Adds the array of this ansatz and the array of another ansatz.
+
+        Args:
+            other: Another ansatz.
+
+        Raises:
+        #TODO: make sure this is the correct type of the wrong shape
+            TypeError: If both of array don't have the same shape.
```

Comment:
I would just put ValueError, if the problem only has to do with the shapes. TypeError is for when I pass the wrong type. In any case, remove the TODO

Reply:
Thanks